libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -107,6 +107,8 @@ void ExpressionCompiler::appendStateVariableInitialization(VariableDeclaration c
 	}
 	if (_varDecl.immutable())
 		ImmutableItem(m_context, _varDecl).storeValue(*type, _varDecl.location(), true);
+	else if (_varDecl.referenceLocation() == VariableDeclaration::Location::Transient)
+		TransientStorageItem(m_context, _varDecl).storeValue(*type, _varDecl.location(), true);
```

Comment:
Is this dead code? Should this be a failing assertion instead after we disallowed initializing transient storage state variables? Will this still generate constructor code to zero-initialize state variables?